custom_components/zwave_mqtt/binary_sensor.py

```diff
@@ -21,26 +22,261 @@
+        NOTIFICATION_TYPE: NOTIFICATION_HOME_SECURITY,
+        NOTIFICATION_VALUES: [3, 4, 9],
+        NOTIFICATION_DEVICE_CLASS: DEVICE_CLASS_SAFETY,
+        NOTIFICATION_SENSOR_ENABLED: True,
+    },
+    {
+        # Index 7: Home Security - Value Id's 5, 6 (glass breakage)
+        NOTIFICATION_TYPE: NOTIFICATION_HOME_SECURITY,
+        NOTIFICATION_VALUES: [5, 6],
+        NOTIFICATION_DEVICE_CLASS: DEVICE_CLASS_SAFETY,
+        NOTIFICATION_SENSOR_ENABLED: True,
+    },
+    {
+        # Index 7: Home Security - Value Id's 7, 8 (motion)
+        NOTIFICATION_TYPE: NOTIFICATION_HOME_SECURITY,
+        NOTIFICATION_VALUES: [7, 8],
+        NOTIFICATION_DEVICE_CLASS: DEVICE_CLASS_MOTION,
+        NOTIFICATION_SENSOR_ENABLED: True,
+    },
+    {
+        # Index 8: Power management - All Value Id's
+        NOTIFICATION_TYPE: NOTIFICATION_POWER_MANAGEMENT,
+        NOTIFICATION_VALUES: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15],
+        NOTIFICATION_DEVICE_CLASS: DEVICE_CLASS_POWER,
+        NOTIFICATION_SENSOR_ENABLED: False,
+    },
+    {
+        # Index 9: System - Value Id's 1, 2, 6, 7
+        NOTIFICATION_TYPE: NOTIFICATION_SYSTEM,
+        NOTIFICATION_VALUES: [1, 2, 6, 7],
+        NOTIFICATION_DEVICE_CLASS: DEVICE_CLASS_PROBLEM,
+        NOTIFICATION_SENSOR_ENABLED: False,
+    },
+    {
+        # Index 10: Emergency - Value Id's 1, 2, 3
+        NOTIFICATION_TYPE: NOTIFICATION_EMERGENCY,
+        NOTIFICATION_VALUES: [1, 2, 3],
+        NOTIFICATION_DEVICE_CLASS: DEVICE_CLASS_PROBLEM,
+        NOTIFICATION_SENSOR_ENABLED: True,
+    },
+    {
+        # Index 11: Clock - Value Id's 1, 2
+        NOTIFICATION_TYPE: NOTIFICATION_CLOCK,
+        NOTIFICATION_VALUES: [1, 2],
+        NOTIFICATION_DEVICE_CLASS: None,
+        NOTIFICATION_SENSOR_ENABLED: False,
+    },
+    {
+        # Index 12: Appliance - All Value Id's
+        NOTIFICATION_TYPE: NOTIFICATION_APPLIANCE,
+        NOTIFICATION_VALUES: [
+            1,
+            2,
+            3,
+            4,
+            5,
+            6,
+            7,
+            8,
+            9,
+            10,
+            11,
+            12,
+            13,
+            14,
+            15,
+            16,
+            17,
+            18,
+            19,
+            20,
+            21,
+        ],
+        NOTIFICATION_DEVICE_CLASS: None,
+        NOTIFICATION_SENSOR_ENABLED: True,
+    },
+    {
+        # Index 13: Home Health - Value Id's 1,2,3,4,5
+        NOTIFICATION_TYPE: NOTIFICATION_APPLIANCE,
+        NOTIFICATION_VALUES: [1, 2, 3, 4, 5],
+        NOTIFICATION_DEVICE_CLASS: None,
+        NOTIFICATION_SENSOR_ENABLED: True,
+    },
+    {
+        # Index 14: Siren
+        NOTIFICATION_TYPE: NOTIFICATION_SIREN,
+        NOTIFICATION_VALUES: [1],
+        NOTIFICATION_DEVICE_CLASS: DEVICE_CLASS_SOUND,
+        NOTIFICATION_SENSOR_ENABLED: True,
+    },
+    {
+        # Index 15: Water valve
+        # ignore non-boolean values
```

Comment:
Wouldn't "Valve current alarm" be an on or off value, with extra attributes for the event data? Or are you thinking the event data reflects a non-binary sensor value?

Reply:
See my previous comment. I have no clue how this looks like in the data structure.

Reply:
My confusion was the comment "non-boolean values". They appear to be boolean (is there an alarm, yes or no), with extra event data to describe the alarm.

If you are ignoring those because you don't know what the event data is, then that's fine.

Reply:
Yes, that is correct. I do not know the data structure that will be send to MQTT for these kind of values.